FifthLesson/VectorProgram/VectorProgram/Model/Vector.cs

```diff
@@ -1,115 +1,137 @@
         }
 
         public static Vector operator *(Vector vector, double numeric)
         {
             return numeric * vector;
         }
 
-        public static Vector operator *(Vector vector1, Vector vector2)
+        public static Vector operator *(Vector first, Vector second)
         {
-            return new Vector(vector1.CoordinateFirst * vector2.CoordinateFirst, vector1.CoordinateSecond * vector2.CoordinateSecond, vector1.CoordinateThird * vector2.CoordinateThird);
+            return new Vector(first.FirstCoordinate * second.FirstCoordinate, first.SecondCoordinate * second.SecondCoordinate, first.ThirdCoordinate * second.ThirdCoordinate);
         }
 
-        public static bool operator ==(Vector vector1, Vector vector2)
-        {
-            var result = (vector1.CoordinateFirst == vector2.CoordinateFirst && vector1.CoordinateSecond == vector2.CoordinateSecond && vector1.CoordinateThird == vector2.CoordinateThird);
-            return result;
-        }
+        public static bool operator ==(Vector first, Vector second) => Equals(first, second);
+
+        public static bool operator !=(Vector first, Vector second) => !Equals(first, second);
+        #endregion
 
-        public static bool operator !=(Vector vector1, Vector vector2)
+        public virtual bool Equals(Vector other)
         {
-            return !(vector1 == vector2);
+            if ((object)this == (object)other) return true;
+            if ((object)other == null) return false;
+            if (GetType() != other.GetType()) return false;
+            return EqualsHelper(this, other);
         }
 
+        public static bool Equals(Vector first, Vector second) => first?.Equals(second) ?? (object)first == (object)second;
+
         public double AngleBetweenVectors(Vector vector)
         {
-            var dividend = CoordinateFirst * vector.CoordinateFirst + CoordinateSecond * vector.CoordinateSecond + CoordinateThird + vector.CoordinateThird;
-            var divisor1 = Math.Sqrt(CoordinateFirst * CoordinateFirst + CoordinateSecond * CoordinateSecond + CoordinateThird * CoordinateThird);
-            var divisor2 = Math.Sqrt(vector.CoordinateFirst * vector.CoordinateFirst + vector.CoordinateSecond * vector.CoordinateSecond + vector.CoordinateThird * vector.CoordinateThird);
-            return Math.Round(dividend / (divisor1 * divisor2), 3);
+            var angle = VectorHelper.AngleBetweenVectors(this, vector);
+            return Math.Round(angle, 3);
         }
 
-        public override bool Equals(object obj)
+        public static string VectorMultiplicate(Vector first, Vector second)
         {
-            return base.Equals(obj);
+            return VectorHelper.VectorMultiplicate(first, second);
         }
 
-        public override int GetHashCode()
+        protected static int GetHashCodeHelper(int[] hashCodeProperties)
```

Comment:
For both helpers:
**Helper** - is an object. **Helper** could be used in class name.